daemons/attrd/attrd_cib.c

```diff
@@ -593,6 +709,11 @@ write_attribute(attribute_t *a, bool ignore_delay)
 
         // Defer write if this is a cluster node that's never been seen
         if (uuid == NULL) {
+            if (pcmk_is_set(v->flags, attrd_value_removed)) {
+                crm_debug("Cannot remove %s[%s] from CIB because "
+                          "node's UUID is unknown", a->id, v->nodename);
+                continue;
+            }
```

Comment:
The whole point of the removed flag is we need to clear it from the CIB (which happens whenever writing out a NULL value). Previously, the controller removed it from the CIB, and we just ignored that in the attribute manager. Now, we want to do it in the attribute manager, but there may not be a writer when it needs to be cleared (in fact that's the main problem case), so we instead mark it and clear it from the CIB once we have a writer. As a bonus we're not dropping the value from memory until we get the confirmation that the CIB change succeeded (rather than just dropping it when we request the CIB change, since all nodes will need to drop it, not just the writer).

Reply:
I'm not sure what the chances are that we can get here for a removed attribute. I'm hoping to get a better idea during testing. I believe it won't be a problem in normal operation for the reasons listed in the earlier comment.